### PR TITLE
Adding Crispy-Hud for Hexen and align Heretic Crispy-Hud

### PR DESCRIPTION
**Pull request by Noseey**
Related Issues:
Closes https://github.com/fabiangreffrath/crispy-doom/issues/1235
Closes https://github.com/fabiangreffrath/crispy-doom/issues/656

**Changes Summary**

Adding Crispy-Hud for Hexen:
- Added Screenblock 12 for new optional crispy hud.
- Limit screenblock inputs to max. 11 for view sizes and draw sliders.
- Modified DrawFullScreenstuff to distinguish based on screenblock if crispy-hud or vanilla fullscreen hud.
- Implemented crispy-hud in DrawFullScreenstuff aligning with aspect ratio.
- Dont' draw Weapon-Pieces and Mana when Gem1 is overlapping with Mana.
- Fixing Item alignment inside the Item Bar to match Standard Hud, also adjusted PatchSELECTBOX to be correct.

Aligning Crispy-Hud Heretic with Hexen:
- Use height for Item Selector Box based upon the fullscreen vanilla HUD, not the Standard-Hud (same as in Hexen).
- Dont' draw Ammo and Jewels when Gem2 is overlapping with Jewels (same as in Hexen).
- Fix Position of Gem2 to match Standard Hud.

### CODE REVIEW

**Comment by Noseey** [comment]
Do you happen to know how I can clean-up the commit-history to only show the recent changes and ignore the ones already merged before?

File: src/hexen/sb_bar.c [hunk 1]
```diff
@@ -1557,14 +1557,14 @@ void DrawFullScreenStuff(void)
                 if (CPlayer->inventorySlotNum > x + i
                     && CPlayer->inventory[x + i].type != arti_none)
                 {
-                    V_DrawPatch(49 + i * 31, 167,
+                    V_DrawPatch(48 + i * 31, 167,
```
[the remark this answers is not in the document]
Comment (Noseey) [reply src/hexen/sb_bar.c:1560]: > Oh, I was just skimming through this PR today and wondered if it was finished or if I am still missing something. 😁 Now, where do these "magic numbers" come from?

It was at first, but I recently checked the Heretic one and decided to add two commits, one of which you already saw. Please bear with me, I will add the second one in a moment and then you can do the full review. ;)

[the remark this answers is not in the document]
Comment (Noseey) [reply src/hexen/sb_bar.c:1560]: Now it is ready for review, sorry for that!

[the remark this answers is not in the document]
Comment (Noseey) [reply src/hexen/sb_bar.c:1560]: > Now, where do these "magic numbers" come from?

Just realized that I didn't answer this question. The change is done for the re-alignment of the Item within the Item Bar Sprite. I took the initial positions from the vanilla fullscreen hud, but noticed that they are misaligned in regards to the Item Position and Small-Number Position. Here the examples:

1) Vanilla (non-fullscreen) Hud item with correct alignment:
Item_Standard_Hud

2) Vanilla (fullscreen) Hud item with wrong alignment: (Item too far to the right, small number too far up)
Item_Fullscreen_hud

3) Crispy Hud item with (now) correct alignment matching Vanilla non-fullscreen Hud:
Item_Crispy_Hud

However, what I just realized today is that the selector for the Item is currently still misaligned by 1px:

1) Vanilla (non-fullscreen) Hud Selector:
Item_Selector_Standard

2) Crispy Hud Selector
Item_Selector_Crispy

If you want, I can bring in a fix for that aswell. Then I will definitely stop the pixel peeping!

[the remark this answers is not in the document]
Comment (fabiangreffrath) [reply src/hexen/sb_bar.c:1560]: Yes, please bring everything in alignment so we can merge this.

[the remark this answers is not in the document]
Comment (Noseey) [reply src/hexen/sb_bar.c:1560]: > Yes, please bring everything in alignment so we can merge this.

Alright, the fix is in now. Well, wouldn't you know, it was actually 2px. ;)
Item_Selector_Crispy_corrected

**Review by fabiangreffrath — APPROVED** [review]
Thank you!

**Pull request merged**